README.md

```diff
@@ -1,349 +1,449 @@
+})
+
+app.post('/create-account', createAccountLimiter, (request, response) => {
+	//...
+})
 ```
 
-**Note:** most stores will require additional configuration, such as custom prefixes, when using multiple instances. The default built-in memory store is an exception to this rule.
+> **Note:** most stores will require additional configuration, such as custom
+> prefixes, when using multiple instances. The default built-in memory store is
+> an exception to this rule.
 
 ## Request API
 
-A `req.rateLimit` property is added to all requests with the `limit`, `current`, and `remaining` number of requests and, if the store provides it, a `resetTime` Date object. These may be used in your application code to take additional actions or inform the user of their status.
+A `request.rateLimit` property is added to all requests with the `limit`,
+`current`, and `remaining` number of requests and, if the store provides it, a
+`resetTime` Date object. These may be used in your application code to take
+additional actions or inform the user of their status.
 
-The property name can be configured with the configuration option `requestPropertyName`
+The property name can be configured with the configuration option
+`requestPropertyName`
 
 ## Configuration options
 
-### max
+### `windowMs`
+
+Time frame for which requests are checked/remembered. Also used in the
+`Retry-After` header when the limit is reached.
+
+Note: with non-default stores, you may need to configure this value twice, once
+here and once on the store. In some cases the units also differ (e.g. seconds vs
+miliseconds)
 
-Max number of connections during `windowMs` milliseconds before sending a 429 response.
+Defaults to `60000` ms (= 1 minute).
 
-May be a number, or a function that returns a number or a promise. If `max` is a function, it will be called with `req` and `res` params.
+### `max`
+
+Max number of connections during `windowMs` milliseconds before sending a 429
+response.
+
+May be a number, or a function that returns a number or a promise. If `max` is a
+function, it will be called with `request` and `response` params.
 
 Defaults to `5`. Set to `0` to disable.
 
 Example of using a function:
 
 ```js
-const rateLimit = require("express-rate-limit");
+import rateLimit from 'express-rate-limit'
 
-function isPremium(req) {
-  //...
+const isPremium = (request) => {
+	// ...
 }
 
 const limiter = rateLimit({
-  windowMs: 15 * 60 * 1000, // 15 minutes
-  
-  // max could also be an async function or return a promise
-  max: function(req, res) {
-    if (isPremium(req)) {
-      return 10;
-    }
-    return 5;
-  }
-});
-
-//  apply to all requests
-app.use(limiter);
+	windowMs: 15 * 60 * 1000, // 15 minutes
+	// `max` could also be an async function or return a promise
+	max: (request, response) => {
+		if (isPremium(request)) return 10
+		else return 5
+	},
+	standardHeaders: true, // Return rate limit info in the `RateLimit-*` headers
```

Comment:
Lets omit the other config options here, the point of this is to provide an example of using the `max` option.

```suggestion
	// `max` could also be an async function or return a promise
	max: (request, response) => {
		if (isPremium(request)) return 10
		else return 5
	},
	// ...
```